server/game/ChessClock.js

```diff
@@ -0,0 +1,98 @@
+class ChessClock {
+    constructor(player, time, delayToStartClock) {
+        this.player = player;
+        this.mainTime = time * 60;
+        this.timeLeft = time * 60;
+        this.mode = 'inactive';
+        this.timerStart = 0; //when the clock is running, this contains the instant on which it started
+        this.paused = false; //has the clock been paused via chat command?
+        this.running = false; //is the clock ticking for the player?
+        this.stateId = 0;
+        this.name = 'ChessClock';
+        this.activated = false; //indicates if the game has started/setup is finished 

```

Comment:
Is this rule (no clocks until after set up) consistent with the current time limit clock?

Reply:
yes. "normal" time limit is started after setup is finished